engine/runtime/src/test/java/org/enso/compiler/SerializationManagerTest.java

```diff
@@ -0,0 +1,110 @@
+package org.enso.compiler;
+
+import com.oracle.truffle.api.TruffleFile;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import org.apache.commons.io.FileUtils;
+import org.enso.editions.LibraryName;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.util.TruffleFileSystem;
+import org.enso.interpreter.test.InterpreterContext;
+import org.enso.pkg.Package;
+import org.enso.pkg.PackageManager;
+import org.enso.polyglot.LanguageInfo;
+import org.enso.polyglot.MethodNames;
+import org.enso.polyglot.Suggestion;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SerializationManagerTest {
+
+  private final PackageManager<TruffleFile> packageManager;
+  private final InterpreterContext interpreterContext;
+  private final EnsoContext ensoContext;
+
+  public SerializationManagerTest() {
+    packageManager = new PackageManager<>(new TruffleFileSystem());
+    interpreterContext = new InterpreterContext(x -> x);
+    ensoContext =
+        interpreterContext
+            .ctx()
+            .getBindings(LanguageInfo.ID)
+            .invokeMember(MethodNames.TopScope.LEAK_CONTEXT)
+            .asHostObject();
+  }
+
+  private Path getLibraryPath(LibraryName libraryName) {
+    return Paths.get(
+            interpreterContext.languageHome(),
+            "..",
+            "lib",
+            libraryName.namespace(),
+            libraryName.name(),
+            interpreterContext.edition())
+        .toAbsolutePath()
+        .normalize();
+  }
+
+  private Package<TruffleFile> getLibraryPackage(LibraryName libraryName) {
+    Path libraryPath = getLibraryPath(libraryName);
+    TruffleFile libraryFile =
+        ensoContext.getEnvironment().getInternalTruffleFile(libraryPath.toString());
+    return packageManager.loadPackage(libraryFile).get();
+  }
+
+  private void clearLibraryCache(LibraryName libraryName) {
+    Path libraryCachePath = Paths.get(getLibraryPath(libraryName).toString(), ".enso");
+    FileUtils.deleteQuietly(libraryCachePath.toFile());
+  }
+
+  @Test
+  public void serializeLibrarySuggestions()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    SerializationManager serializationManager = new SerializationManager(ensoContext.getCompiler());
+    LibraryName standardBaseLibrary = new LibraryName("Standard", "Base");
+    Package<TruffleFile> standardBasePackage = getLibraryPackage(standardBaseLibrary);
+    ensoContext
+        .getPackageRepository()
+        .registerMainProjectPackage(standardBaseLibrary, standardBasePackage);
+    Object result = ensoContext.getCompiler().compile(false, false).get(20, TimeUnit.SECONDS);
```

Comment:
Let's have this as a constant somewhere. Easier to spot than deep within the test implementation.